diff --git a/cpp/core/base/KAGParser.cpp b/cpp/core/base/KAGParser.cpp
--- a/cpp/core/base/KAGParser.cpp
+++ b/cpp/core/base/KAGParser.cpp
@@ -1386,6 +1386,7 @@ iTJSDispatch2 *tTJSNI_KAGParser::_GetNextTag() {
     static ttstr __storage_name(TJSMapGlobalStringMap(TJS_W("storage")));
     static ttstr __target_name(TJSMapGlobalStringMap(TJS_W("target")));
     static ttstr __exp_name(TJSMapGlobalStringMap(TJS_W("exp")));
+    static ttstr __escape_name(TJSMapGlobalStringMap(TJS_W("escape")));
 
     // here, processes one line
     while(true) {
@@ -1791,11 +1792,17 @@ iTJSDispatch2 *tTJSNI_KAGParser::_GetNextTag() {
                             TVPExecuteExpression(exp, Owner, &val);
                             exp = val;
 
+                            bool escape = true;
+                            if(TJS_SUCCEEDED(DicObj->PropGet(
+                                   TJS_MEMBERMUSTEXIST, __escape_name.c_str(),
+                                   __escape_name.GetHint(), &val, DicObj))) {
+                                escape = val.operator bool();
+                            }
+
                             // count '['
                             const tjs_char *p = exp.c_str();
                             tjs_int r_count = 0;
                             while(*p) {
-                                if(*p == TJS_W('['))
+                                if(escape && *p == TJS_W('['))
                                     r_count++;
                                 p++;
                                 r_count++;
@@ -1818,7 +1825,7 @@ iTJSDispatch2 *tTJSNI_KAGParser::_GetNextTag() {
                             // escape '['
                             p = exp.c_str();
                             while(*p) {
-                                if(*p == TJS_W('[')) {
+                                if(escape && *p == TJS_W('[')) {
                                     *d = TJS_W('[');
                                     d++;
                                     *d = TJS_W('[');
