diff --git a/cpp/core/environ/RenderUtils.h b/cpp/core/environ/RenderUtils.h
index 3c64968..ac13d13 100644
--- a/cpp/core/environ/RenderUtils.h
+++ b/cpp/core/environ/RenderUtils.h
@@ -6,9 +6,21 @@
 #include <cocos2d.h>
 #include "Platform.h"
 
-static inline void (*_postUpdate)() = nullptr;
+using TVPPostUpdateEvent = void (*)();
 
-inline void TVPSetPostUpdateEvent(void (*f)()) { _postUpdate = f; }
+inline TVPPostUpdateEvent &TVPGetPostUpdateEvent() {
+    static TVPPostUpdateEvent event = nullptr;
+    return event;
+}
+
+inline void TVPSetPostUpdateEvent(TVPPostUpdateEvent event) {
+    TVPGetPostUpdateEvent() = event;
+}
+
+inline void TVPRunPostUpdateEvent() {
+    if(TVPPostUpdateEvent event = TVPGetPostUpdateEvent())
+        event();
+}
 
 inline int TVPDrawSceneOnce(int interval) {
     static tjs_uint64 lastTick = TVPGetRoughTickCount32();
@@ -16,8 +28,7 @@ inline int TVPDrawSceneOnce(int interval) {
     tjs_uint64 curTick = TVPGetRoughTickCount32();
     int remain = interval - (curTick - lastTick);
     if(remain <= 0) {
-        if(_postUpdate)
-            _postUpdate();
+        TVPRunPostUpdateEvent();
         auto *director = cocos2d::Director::getInstance();
         director->drawScene();
         TVPForceSwapBuffer();
diff --git a/cpp/core/environ/cocos2d/MainScene.cpp b/cpp/core/environ/cocos2d/MainScene.cpp
index 8b2dc0d..681402a 100644
--- a/cpp/core/environ/cocos2d/MainScene.cpp
+++ b/cpp/core/environ/cocos2d/MainScene.cpp
@@ -897,8 +897,7 @@ void TVPMainScene::update(float delta) {
     // TVPWindowLayer::_currentWindowLayer->UpdateOverlay();
     iTVPTexture2D::RecycleProcess();
     //_ResotreGLStatues();
-    if(_postUpdate)
-        _postUpdate();
+    TVPRunPostUpdateEvent();
     if(_fpsLabel) {
         unsigned int drawCount;
         uint64_t vmemsize;
